TeamCode/src/main/java/org/firstinspires/ftc/teamcode/RobotController.java

```diff
@@ -0,0 +1,64 @@
+package main.java.org.firstinspires.ftc.teamcode;
+
+import java.util.List;
+import java.util.ArrayList;
+import layer.LayerSetUpInfo;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+import java.util.function.Consumer;
+// import task interface and what not
+// Some of the imports above may have different names after translation.
+
+class RobotController{
+
+    LayerSetUpInfo layerSetupInfo;
+    List<Consumer<Boolean>> updateListeners;
+    List<Object<? extends Layer>> layerObjects; // Added layerObjects as a field
+
+    // Swap robot with what the variable is actually named.
+    RobotController(HardwareMap robot){
+        this.layerSetupInfo = new LayerSetUpInfo(robot);
+        this.updateListeners = new ArrayList<>();
+        this.layerObjects = new ArrayList<>();
+    }
+
+    void setup(List<Class<? extends Layer>> capCObjects){
+         for(Class<? extends Layer> items: capCObjects){
+            this.layerObjects.add(items.getConstructor(LayerSetUpInfo.class).newInstance(LayerSetUpInfo));
+         }
+    }
+
+    boolean update(){
+            // Call all update listeners
+        for(Consumer<Boolean> listener: updateListeners){
+            listener(false);
+        }
+            // Do work on layers
+            int i = 0;
+            // Possible way to implement sentinel? could be wrong, and I just realized that task is actually a marker interface now.
+            Object<? extends task> currentTask = null; 
```

Comment:
change Object<? extends task> to Object<? extends Task>. I forgot to put the first letter in uppercase to match the interface.